src/auth/auth.service.ts

```diff
@@ -91,4 +91,13 @@ export class AuthService {
     const user = await this.getUserFromHeaders(headers);
     return user.id;
   };
+
+  getOptionalUserIdFromHeader = async (headers: Headers) => {
+    try {
+      const user = await this.getUserFromHeaders(headers);
+      return user.id;
+    } catch (error) {
+      return null;
+    }
+  };
```

Comment:
```suggestion
  getOptionalUserIdFromHeader = async (headers: Headers) => {
    try {
      const user = await this.getUserFromHeaders(headers);
      return user.id;
    } catch (error) {
      return null;
    }
  };
```
This feels like a slippery - masking any error with a `null`.
We should either emphasis to avoid using this function whenever possible using a jsdoc, or catch a more specific error.

Reply:
Agree, added check for only AuthenticationError